src/server/client.rs

```diff
@@ -223,9 +223,50 @@ async fn handle_autoalloc_message(
         AutoAllocRequest::AddQueue(params) => create_queue(server_dir, state_ref, params).await,
         AutoAllocRequest::Events { descriptor } => get_event_log(state_ref, descriptor),
         AutoAllocRequest::Info { descriptor } => get_allocations(state_ref, descriptor),
+        AutoAllocRequest::RemoveQueue(name) => remove_queue(state_ref, name).await,
     }
 }
 
+async fn remove_queue(state_ref: &StateRef, id_to_delete: DescriptorId) -> ToClientMessage {
+    let futures: Vec<_> = {
```

Comment:
This seems to be quite complicated just to get a single descriptor. Just use `get_descriptor_mut` and if it returns `None`, return an error.